src/tools/rbd_nbd/rbd-nbd.cc

```diff
@@ -466,19 +484,19 @@ class NBDWatchCtx : public librbd::UpdateWatchCtx
   }
 };
 
-static int open_device(const char* path, bool try_load_module = false)
+static int open_device(const char* path, Config *cfg = NULL, bool try_load_module = false)
```

Comment:
The default to NULL looks like unnecessary and actually it will crash with this default.